backend/src/api/routers/organizations.py

```diff
@@ -0,0 +1,178 @@
+    "e.V. - Eingetragener Verein",
+    "gGmbH - Gemeinnützige Gesellschaft mit beschränkter Haftung",
+    "GmbH - Gesellschaft mit beschränkter Haftung",
+    "gUG - Gemeinnützige Unternehmergesellschaft",
+    "UG - Unternehmergesellschaft",
+    "Stiftung",
+]
+
+SECTORS = Literal[
+    "Bildung",
+    "Gesundheit",
+    "Kultur",
+    "Sport",
+    "Umwelt",
+]
+
+
+async def get_organization_by_id(
+    db_session: Annotated[AsyncSession, Depends(get_async_session)],
+    organization_id: Annotated[UUID, Path()],
+) -> Organization:
+    organization = await db_session.get(Organization, organization_id)
+    if organization is None:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, "Organization not found.")
+    return organization
+
+
+class OrganizationContactRead(BaseModel):
+    model_config = ConfigDict(from_attributes=True)
+
+    name: str = Field(default=...)
+    role: str = Field(default=...)
+    email: str = Field(default=...)
+    phone: str = Field(default=...)
+
+
+class OrganizationRead(BaseModel):
+    model_config = ConfigDict(from_attributes=True)
+
+    id: UUID = Field(default=...)
+    name: str = Field(default=...)
+    legal_form: LEGAL_FORMS = Field(default=...)
+    sectors: list[SECTORS] = Field(default=...)
+    contacts: list[OrganizationContactRead] = Field(default=[])
+    created_at: datetime = Field(default=...)
+    created_by: str = Field(default=...)
+    updated_at: datetime = Field(default=...)
+    updated_by: str = Field(default=...)
+    archived_at: datetime | None = Field(default=...)
+    archived_by: str | None = Field(default=...)
+
+    @model_validator(mode="before")
+    @classmethod
+    def set_user_names(cls, data: Any) -> Any:
+        data = copy(x=data)
+        data.created_by = data.creator.name
+        data.updated_by = data.updater.name
+        data.archived_by = data.archiver.name if data.archiver else None
+        return data
+
+
+class OrganizationCreate(BaseModel):
+    name: str = Field(default=...)
+    legal_form: LEGAL_FORMS = Field(default=...)
+    sectors: list[SECTORS] = Field(default=...)
+    contacts: list[OrganizationContactRead] = Field(default=[])
+
+
+@router.post(path="", response_model=OrganizationRead)
+async def create_organization(
+    organization: OrganizationCreate,
+    current_active_user: Annotated[User, Depends(current_active_user)],
+    db_session: Annotated[AsyncSession, Depends(get_async_session)],
+) -> Organization:
+    new_organization = Organization(
+        id=uuid4(),
+        name=organization.name,
+        legal_form=organization.legal_form,
+        created_by=current_active_user.id,
+        updated_by=current_active_user.id,
+        sectors=organization.sectors,
+        contacts=[
+            OrganizationContact(**contact.model_dump())
+            for contact in organization.contacts
+        ],
+    )
+    db_session.add(instance=new_organization)
+    await db_session.commit()
+    return new_organization
```

Comment:
I'm getting an error inside the model validation here. The error is not very helpful and is

> raise exc.MissingGreenlet(
engage-2-api-1       | sqlalchemy.exc.MissingGreenlet: greenlet_spawn has not been called; can't call await_only() here. Was IO attempted in an unexpected place? (Background on this error at: https://sqlalche.me/e/20/xd2s)

Inside the stack trace it seems to relate to the custom model validation
> File "/app/api/routers/organizations.py", line 74, in set_user_names
engage-2-api-1       |     data.created_by = data.creator.name

Since the endpoint has the explicit commit above, the data still ends up in the database even though the request returns 500.

Reply:
For instance the following payload creates produces the issue.

```python
{
  "name": "Do Good e.V.",
  "legal_form": "e.V. - Eingetragener Verein",
  "sectors": [
    "Bildung"
  ],
  "contacts": [
		{"name": "John Doe",
		"role" : "Important Person",
		"email" : "John@doe.com",
		"phone" : "32391425"}
	]
}
```